driver.go

```diff
@@ -702,8 +708,8 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 
 	cleanup := func() {
 		d.logger.Debug("Cleaning up", "container", containerID)
-		if err = d.podman.ContainerDelete(d.ctx, containerID, true, true); err != nil {
-			d.logger.Error("failed to clean up from an error in Start", "error", err)
+		if cleanupErr := d.podman.ContainerDelete(d.ctx, containerID, true, true); cleanupErr != nil {
+			d.logger.Error("failed to clean up from an error in Start", "error", cleanupErr)
```

Comment:
fun fact - this would overwrite `err` during the `cleanup()` call, causing what is now `startErr` to always be nil, which was beyond confusing.

Reply:
Oof..nice catch!